settings/update_settings.go

```diff
@@ -0,0 +1,29 @@
+package settings
+
+import "reflect"
+
+type UpdateSettings struct {
+	Blobstores       []Blobstore      `json:"blobstores"`
+	DiskAssociations DiskAssociations `json:"disk_associations"`
+	Mbus             MBus             `json:"mbus"`
+	TrustedCerts     string           `json:"trusted_certs"`
+}
+
+func (updateSettings *UpdateSettings) MergeSettings(newSettings UpdateSettings) bool {
+	var restartNeeded bool
+
+	updateSettings.TrustedCerts = newSettings.TrustedCerts
+	updateSettings.DiskAssociations = newSettings.DiskAssociations
+
+	if !reflect.DeepEqual(newSettings.Mbus, updateSettings.Mbus) && !reflect.DeepEqual(newSettings.Mbus, MBus{}) {
+		updateSettings.Mbus = newSettings.Mbus
+		restartNeeded = true
```

Comment:
We could rename `restartNeeded` to something like `changed` because `UpdateSettings` shouldn't know anything about restart. A really minor thing.

Reply:
Good call, we made it a big more verbose since `changed` could have been misleading since the trusted certs or disk associations could have changed and this value would have been `false` still.